core/src/plugins/kubernetes/run.ts

```diff
@@ -1237,13 +1239,23 @@ export class PodRunner {
   /**
    * Helper to detect pod disruption, and throw NotFoundError in case the Pod has been evicted
    *
+   * @param lastEventTime Don't just use new Date() as the timezone might differ between client and server. Get the max
+   *        lastEventTime from the events returned by `getResourceEvents` and pass that as the parameter here.
    * @throws NotFoundError
    */
-  private async throwIfPodKilled(afterTime: Date): Promise<void> {
+  private async throwIfPodKilled(lastEventTime: Date | undefined): Promise<void> {
     const events = await getResourceEvents(this.api, this.pod)
     if (
       // If reason is killed and lastTimestamp doesn't exist or is greater than afterTime
-      some(events, (event) => event.reason === "Killing" && (!event.lastTimestamp || event.lastTimestamp > afterTime))
+      some(events, (event) => {
+        // only consider events that happened after the last event we've seen, and if they have a lastTimestamp property.
+        // we ignore all events without lastTimestamp, because that could lead to permanently unusable Pods.
+        if (lastEventTime && event.lastTimestamp) {
+          return event.lastTimestamp > lastEventTime && event.reason === "Killing"
+        }
```

Comment:
I think keeping the time check separate from reason comparison makes the code easier to understand. I think this should have the same effect:

```suggestion
        if (lastEventTime && event.lastTimestamp && event.lastTimestamp > lastEventTime) {
          return false
        }
```

Reply:
Maybe it's more clear if we move the time check to a `.filter` call directly after `getResourceEvents`